### DISCUSSION TITLE
Audit-Script

### DISCUSSION THREAD

**Discussion by tordans** [octoherd/octoherd#9] [Ideas]
Thanks for introducing octoherd at https://www.youtube.com/watch?v=qXHiXSSD3Kc. I think I remember you asked for octoherd script ideas, so here is one:

My usecase: I want to make sure all repos in my organisation have the correct configuration. I could click through all settings screens on all repos – which I do – but maybe this is a octoherd usecase.

Idea: Look at all given repos (orga, user, list). Create a line in a csv per repos. Columns could be common repos settings like "issues on/off", "projects on/off", "gh pages on/off", "gh pages url", etc. maybe also access information.

**Comment by gr2m** [comment]
> Create a line in a csv per repos

Octoherd creates a log file in NDJSON format by default. Everytime you use `octokit.log.{debug,info,warn,error}`, a line is added to the log file, while only the message is logged to std out. You can learn more about it here: https://github.com/octoherd/octokit/#logging. The path to the logfile is currently logged to stdout when the run is done

A super simple script would look like this

```js
/**
 * The "Hello, World!" of all Octoherd Scripts!
 *
 * @param {import('@octoherd/cli').Octokit} octokit
 * @param {import('@octoherd/cli').Repository} repository
 */
export async function script(octokit, repository) {
  octokit.log.info(repository)
}
```

Then you could use [`jq`](https://stedolan.github.io/jq/) or a custom script to turn the log file into a CSV. Maybe only log the relevant information instead of all of the `repository`, it will greatly reduce size of the log file.

Another approach could be to load the repository settings from one repository and then compare them to all others, similar to what https://github.com/octoherd/script-sync-branch-protections

Yet another approach could be to introduce script-specific options for the options you are interested in.

I'm thinking about adding a reports feature to octoherd scripts, but I want to implement the Deno and Web runners first. Octoherd script should work across Node/Deno/Browser, so creating a CSV file directly from the script wouldn't be the Octoherd way, but of course you can do that anyway

I've created an `npm init` script which walks you through creating an Octoherd script and repository: https://github.com/octokit/create-octokit-project.js/, give it a try
